### ISSUE TITLE
Import/Export download links from/to a txt files

### ISSUE THREAD

**Issue by bup4gr** [marklieberman/downloadstar#31]
When you have a txt file that has a download link in every line, automatically imports the links in download star, otherwise you have to manually insert each link.

**Comment by OkanEsen** [comment]
I personally am against this idea, since Download Star imo should stay lightweight and just concentrate on the things it does now. There are a lot of download managers (jDownloader, Turbo Download Manager, etc.) where this feature would be more appropriate to implement, if it wasn't already, but not here.

It's of course the decision of @marklieberman, if he wants to go ahead with this idea, just my two cents.

**Comment by bup4gr** [comment]
For my experience, external download manager has a great limitation: all sites that use identification through cookies or javascript sessions do not work in external download manager.
In these cases, the download does not work.

**Comment by marklieberman** [comment]
Exporting links by serving a data URL containing a list of text, JSON, or CSV is quite trivial. 

However, if you are expecting to get a list of cURL-able links then it gets more complicated. In that case, I'd have to use cookies permission to build the request string with cookies. I'd probably then want to have cookies optional_permission.

Importing links can probably be implemented as part of the UI for #6.

Basically, I'll consider these features as long as it doesn't grow too much in scope.

**Comment by unicorndreams** [comment]
I also would like to see this feature, more than any other! I rarely used DTA to download things from a Page. Instead, I used it to "extract" those links, export them, fiddle with them, them reimport the new links to DTA. That's actually the absolute ONLY reason I keep a Firefox "legacy" installed (running on "-no-remote" mode), because now and then I need DTA to download "external" links for me (not links inside a Page, even though they are related to some).

Currently, "Download Status Bar S3" is doing most of my required random download jobs, as it has a "Create new download" option (of course, when I have to download massive amounts of data, from sites which require user authentication, only Firefox "legacy" and DTA can accomplish the deed).

**Comment by kgbm3** [comment]
> >  bup4gr commented on Jun 19
> 
> When you have a txt file that has a download link in every line, automatically imports the links in download star, otherwise you have to manually insert each link.dfg

Alright guys, call me stupid, bt _how_ do you add a (download) link, manually? o.0

Trying to get the (PDF) file @

`https://universalflowuniversity.com/Books/Computer Programming/Microsoft Windows Development/Troubleshooting with the Windows Sysinternals Tools 2nd Edition.pdf`

2nd Edition, Google search query @

`https://www.google.com/search?q=troubleshooting+with+the+windows+sysinternals+tools+book+download`

^^ really, not trying to be retarded; what needs to be done? :o

** this is what eye see, [nothing](https://ibb.co/f3SPo9): 


![](https://image.ibb.co/eeaoFp/2018_08_11_02_16_14_Troubleshooting_with_the_Windows_Sysinternals_Tools_2nd_Edition_Troubleshootin.png)

**Comment by marklieberman** [comment]
@kgbm3 The feature doesn't exist yet in Download Star. This issue is a feature request to add it.

Also, web extensions don't work in PDF.js because its considered a privileged URL so DS is unlikely to work if you open the popup there.
